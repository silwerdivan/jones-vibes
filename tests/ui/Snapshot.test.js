import { screen } from '@testing-library/dom';
import fs from 'fs';
import path from 'path';

describe('UI Snapshots', () => {
  beforeAll(async () => {
    const html = fs.readFileSync(path.resolve(__dirname, '../../index.html'), 'utf8');
    document.body.innerHTML = html.toString();
    
    // The scripts are loaded as modules, so we need to handle them.
    await import('../../js/app.js');

    // Manually trigger DOMContentLoaded to run the app script
    document.dispatchEvent(new Event('DOMContentLoaded', { bubbles: true, cancelable: true }));
  });

  test('Player 1 panel should match snapshot', () => {
    const player1Panel = document.getElementById('player1-status-panel');
    expect(player1Panel.innerHTML).toMatchSnapshot();
  });

  test('Player 2 panel should match snapshot', () => {
    const player2Panel = document.getElementById('player2-status-panel');
    expect(player2Panel.innerHTML).toMatchSnapshot();
  });

  test('Info panel should match snapshot', () => {
    const infoPanel = document.getElementById('location-info');
    expect(infoPanel.innerHTML).toMatchSnapshot();
  });

  test('Actions panel should match snapshot', () => {
    const actionsPanel = document.getElementById('game-controls');
    expect(actionsPanel.innerHTML).toMatchSnapshot();
  });

  test('Log panel should match snapshot', () => {
    const logPanel = document.getElementById('log');
    expect(logPanel.innerHTML).toMatchSnapshot();
  });
});