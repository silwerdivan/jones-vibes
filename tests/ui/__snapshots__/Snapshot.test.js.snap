// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`UI Snapshots Actions panel should match snapshot 1`] = `
"
                <h2>Actions</h2>
                <div class="action-buttons">
                    <button id="btn-rest-end-turn">Rest / End Turn</button>
                    <button id="btn-travel" class="location-specific">Travel</button>
                    <button id="btn-work-shift" class="location-specific hidden">Work Shift</button>
                    <button id="btn-take-course" class="location-specific hidden">Take Course</button>
                    <button id="btn-buy-item" class="location-specific hidden">Buy Item</button>
                    <button id="btn-deposit" class="location-specific hidden">Deposit</button>
                    <button id="btn-withdraw" class="location-specific hidden">Withdraw</button>
                    <button id="btn-take-loan" class="location-specific hidden">Take Loan</button>
                    <button id="btn-repay-loan" class="location-specific hidden">Repay Loan</button>
                    <button id="btn-buy-car" class="location-specific hidden">Buy Car</button>
                </div>
            "
`;

exports[`UI Snapshots Info panel should match snapshot 1`] = `
"
                <h2>Current Location</h2>
                <p>Current Location: <span id="current-location">Home</span></p>
                <p>Turn: <span id="game-turn">1</span></p>
            "
`;

exports[`UI Snapshots Log panel should match snapshot 1`] = `
"
                <h3>Game Log</h3>
            "
`;

exports[`UI Snapshots Player 1 panel should match snapshot 1`] = `
"
                <h2>Player 1</h2>
                <div class="stats">
                    <div class="stat-item"><span class="stat-label">Cash:</span> <span id="p1-cash" class="stat-value currency">0</span></div>
                    <div class="stat-item"><span class="stat-label">Savings:</span> <span id="p1-savings" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Loan:</span> <span id="p1-loan" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Happiness:</span> <span id="p1-happiness" class="stat-value">50</span></div>
                    <div class="stat-item"><span class="stat-label">Education:</span> <span id="p1-education" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Career:</span> <span id="p1-career" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Car:</span> <span id="p1-car" class="stat-value">No</span></div>
                </div>
                <div class="time-remaining">Time Remaining: <span id="p1-time">24</span> hours</div>
            "
`;

exports[`UI Snapshots Player 2 panel should match snapshot 1`] = `
"
                <h2>Player 2 (AI)</h2>
                <div class="stats">
                    <div class="stat-item"><span class="stat-label">Cash:</span> <span id="p2-cash" class="stat-value currency">0</span></div>
                    <div class="stat-item"><span class="stat-label">Savings:</span> <span id="p2-savings" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Loan:</span> <span id="p2-loan" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Happiness:</span> <span id="p2-happiness" class="stat-value">50</span></div>
                    <div class="stat-item"><span class="stat-label">Education:</span> <span id="p2-education" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Career:</span> <span id="p2-career" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Car:</span> <span id="p2-car" class="stat-value">No</span></div>
                </div>
                <div class="time-remaining">Time Remaining: <span id="p2-time">24</span> hours</div>
            "
`;
